crates/neon/src/types_impl/buffer/types.rs

```diff
@@ -0,0 +1,760 @@
+    where
+        C: Context<'cx>,
+    {
+        let buffer = cx.array_buffer(len * std::mem::size_of::<T>())?;
+        Self::from_array_buffer(cx, buffer, 0, len)
+    }
+}
+
+macro_rules! impl_typed_array {
+    ($name:expr, $typ:ty, $($pattern:pat)|+, $tag:ident$(,)?) => {
+        impl private::Sealed for $typ {}
+
+        impl Binary for $typ {
+            const RAW: TypedArrayType = TypedArrayType::$tag;
+        }
+
+        impl Value for JsTypedArray<$typ> {}
+
+        impl Object for JsTypedArray<$typ> {}
+
+        impl ValueInternal for JsTypedArray<$typ> {
+            fn name() -> String {
+                $name.to_string()
+            }
+
+            fn is_typeof<Other: Value>(env: Env, other: &Other) -> bool {
+                let env = env.to_raw();
+                let other = other.to_raw();
+
+                if unsafe { !sys::tag::is_typedarray(env, other) } {
+                    return false;
+                }
+
+                let info = unsafe { sys::typedarray::info(env, other) };
+
+                matches!(info.typ, $($pattern)|+)
+            }
+        }
+    };
+}
+
+impl_typed_array!("Int8Array", i8, TypedArrayType::I8, I8);
+impl_typed_array!(
+    "Uint8Array",
+    u8,
+    TypedArrayType::U8 | TypedArrayType::U8Clamped,
+    U8,
+);
+impl_typed_array!("Int16Array", i16, TypedArrayType::I16, I16);
+impl_typed_array!("Uint16Array", u16, TypedArrayType::U16, U16);
+impl_typed_array!("Int32Array", i32, TypedArrayType::I32, I32);
+impl_typed_array!("Uint32Array", u32, TypedArrayType::U32, U32);
+impl_typed_array!("Float32Array", f32, TypedArrayType::F32, F32);
+impl_typed_array!("Float64Array", f64, TypedArrayType::F64, F64);
+impl_typed_array!("BigInt64Array", i64, TypedArrayType::I64, I64);
+impl_typed_array!("BigUint64Array", u64, TypedArrayType::U64, U64);
+
+/// The standard JS [`Int8Array`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Int8Array) type.
+///
+/// # Example
+///
+/// ```
+/// # use neon::prelude::*;
+/// use neon::types::buffer::TypedArray;
+///
+/// fn double(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+///     let mut array: Handle<JsInt8Array> = cx.argument(0)?;
+///
+///     for elem in array.as_mut_slice(&mut cx).iter_mut() {
+///         *elem *= 2;
+///     }
+///
+///     Ok(cx.undefined())
+/// }
+/// ```
+pub type JsInt8Array = JsTypedArray<i8>;
+
+/// The standard JS [`Uint8Array`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array) type.
+///
+/// # Example
+///
+/// ```
+/// # use neon::prelude::*;
+/// use neon::types::buffer::TypedArray;
+///
+/// fn double(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+///     let mut array: Handle<JsUint8Array> = cx.argument(0)?;
+///
+///     for elem in array.as_mut_slice(&mut cx).iter_mut() {
+///         *elem *= 2;
+///     }
+///
+///     Ok(cx.undefined())
+/// }
+/// ```
+pub type JsUint8Array = JsTypedArray<u8>;
```

Comment:
It would be nice to generate the aliases and examples in the `impl_typed_array` macro.